pkg/storage/cache.go

```diff
@@ -0,0 +1,84 @@
+//go:generate mockgen -source cache.go -destination ../../internal/mocks/mock_cache.go -package mocks cache
+
+package storage
+
+import (
+	"time"
+
+	"github.com/karlseguin/ccache/v3"
+)
+
+const defaultMaxCacheSize = 10000
+const defaultCacheTTL = 10 * time.Second
+
+// InMemoryCache is a general purpose cache to store things in memory.
+type InMemoryCache[T any] interface {
+
+	// Get If the key exists, returns the value and "true".
+	// If the key didn't exist, returns nil, false.
+	Get(key string) (*CachedResult[T], bool)
```

Comment:
(non-blocking/ergonomic) Is it possible for this function to ever return a non-nil value and `false`? If not, simply returning `nil` could qualify as sufficient information to infer that the key did not exist in the cache.

Reply:
I tried to replicate the syntax used by the standard map type: https://go.dev/blog/maps

> A two-value assignment tests for the existence of a key:
> 
> i, ok := m["route"]
> 
> In this statement, the first value (i) is assigned the value stored under the key "route". If that key doesn’t exist, i is the value type’s zero value (0). The second value (ok) is a bool that is true if the key exists in the map, and false if not.
> 
> To test for a key without retrieving the value, use an underscore in place of the first value:
> 
> _, ok := m["route"]
>

Reply:
I'll apply the suggestion :)